cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterController.java

```diff
@@ -201,13 +215,18 @@ private void recreateConfigMapWatch() {
     /**
       Periodical reconciliation (in case we lost some event)
      */
-    private void reconcile() {
-        kafkaClusterOperations.reconcileAll(namespace, labels);
-        kafkaConnectClusterOperations.reconcileAll(namespace, labels);
+    private Future<?> reconcile() {
```

Comment:
Why are we changing all the various `reconcile()` to return `Future`, rather than accept `Handler<AsyncResult>`? Do we prefer a `Future`-based API?

Reply:
@tombentley Can't speak for @ppatierno , but didn't you changed most of the APIs from handlers to _return Future_ in the `*Operations` classes? Now I'm now kind of just trying to follow that approach in my code. :-)

Reply:
Taking a look at the code before writing this down, it seemed to me that the handlers approach would have been a mess (a simple example, the `reconcile` in the CC calls the three `reconcileAll` operations and having `Future`(s) to join is simpler). Another point is that from what I know in the Vert.x community there is a discussion about these two approaches and it seems that the future "could" be more .... `Future`(s). I read that the Scala support is heavily based on that then handlers. In conclusion, using `Future`(s) was the simpler and more reasonable approach to me.

Reply:
FTR, I wasn't saying that `Future`-based API was wrong, I was just trying to understand the pros and cons of doing it that way. I certainly agree that using `join` makes things simpler, but we can do that "inside" the methods, setting the handler on the joined Future to be the `handler` parameter, so it appears to be something where we can choose which style of methods we prefer. I'm still not sure I understand the pros and cons though.